app/views/play/level/tome/SpellView.coffee

```diff
@@ -106,6 +108,15 @@ module.exports = class SpellView extends CocoView
     $(@ace.container).find('.ace_gutter').on 'click', @onGutterClick
     @initAutocomplete aceConfig.liveCompletion ? true
 
+    # Create a Spade to 'dig' into Ace.
+    @spade = new Spade()
+    @spade.track(@ace)
+
+    # If a user is taking longer than 10 minutes, let's log it.
```

Comment:
Let's `return if @session.get('creator') isnt me.id or @session.fake` so that we don't try to save sessions when observing them or when a teacher is playing without saving a real session.